syntax = "proto3";

package transaction_verification;

service TransactionVerification {
  rpc VerificationItems (VIRequest) returns (VerificationResponse);
  rpc VerificationUser (VURequest) returns (VerificationResponse);
  rpc VerificationCreditCard (VCCRequest) returns (VerificationResponse);
}

message Item {
  string name = 1;
  int32 quantity = 2;
}

message Book {
  string bookId = 1;
  string title = 2;
  string author = 3;
}

message User {
  string name = 1;
  string contact = 2;
}

message CreditCard {
  string number = 1;
  string expirationDate = 2;
  string cvv = 3;
}

message VIRequest {
  string orderId = 1;
  repeated Item items = 2;
}

message VCCRequest {
  string orderId = 1;
  CreditCard creditCard = 2;
}

message VURequest {
  string orderId = 1;
  User user = 2;
}

message VerificationResponse {
  string orderId = 1;
  bool verified = 2;
}

message ErrorResponse {
  string code = 1;
  string message = 2;
}